src/main/scala/uk/gov/hmrc/ct/computations/Validators/ComputationValidatableBox.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.ct.computations.Validators
+
+import uk.gov.hmrc.ct.box.retriever.BoxRetriever
+import uk.gov.hmrc.ct.box.{CtOptionalInteger, CtValidation, ValidatableBox}
+import uk.gov.hmrc.ct.computations.retriever.ComputationsBoxRetriever
+
+trait ComputationValidatableBox[T <: BoxRetriever] extends ValidatableBox[T] {
+
+  def fieldhasValueWhenTrading(retriever: ComputationsBoxRetriever, boxId: String, value: Option[Int]) =
```

Comment:
I'd rename this to mandatoryIfCompanyTrading or something similar, bit more explicit